src/java.base/share/classes/java/lang/invoke/DelegatingMethodHandle.java

```diff
@@ -189,5 +189,5 @@ private static Kind whichKind(int whichCache) {
     }
 
     /* Placeholder class for DelegatingMethodHandles generated ahead of time */
-    final class Holder {}
+    static final class Holder {}
```

Comment:
For the four `Holder` classes in `java.lang.invoke`, the class is generated when running jlink via `java.lang.invoke.GenerateJLIClassesHelper`. To stay in sync with the definition you'd have to update that code. Also, since these `Holder` classes will only contain static methods and are never instantiated then I'm not sure it matters whether the classes are static or not.

Reply:
I'll just revert them

Reply:
For static methods, since in java language you cannot declare static method in instance inner classes, I'd say making them static makes more sense language-wise. Also making them static reduces compiler synthetic instance field and constructors.

Reply:
Incidentally, Java-the-language allows static methods in inner instance classes since JDK 16. And I'm not sure this was ever a restriction at the JVMS level since we've been generating static methods into these inner instance classes since at least JDK 9.

Reply:
Inner classes doesn't really exist for the JVM, it's just an attribute (in fact, a pair of attributes) that is read/write by javac (it's very similar to the way generics work).
So it is Ok to have static methods in inner classes since Java 1.1 from the JVM POV with the caveat that you may not be all to call them from Java-the-language.
Also since Java 11, inner classes are also nestmate and those attributes (NestHost/NestMembers) change the behavior of the VM.